core/controllers/payload_validator.py

```diff
@@ -0,0 +1,346 @@
+# coding: utf-8
+
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validates handler args against its schema by calling schema utils.
+Also contains a list of handler class names which does not contain the schema.
+"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import schema_utils
+
+from typing import Any, Dict, List, Text, Tuple # isort:skip  pylint: disable= wrong-import-order, wrong-import-position, unused-import, import-only-modules
+
+
+def validate(handler_args, handler_args_schemas, allowed_extra_args):
+    # type: (Any, Any, bool) -> Tuple[Dict[Any, Any], List[Text]]
+
+    """Calls schema utils for normalization of object against its schema
+    and collects all the errors.
+
+    Args:
+        handler_args: *. Object for normalization.
+        handler_args_schemas: dict. Schema for args.
+        allowed_extra_args: bool. Whether extra args are allowed in handler.
+
+    Returns:
+        *. A two tuple, where the first element represents the normalized value
+        in dict format and the second element represents the lists of errors
+        after validation.
+    """
+    # Collect all errors and present them at once.
+    errors = []
+    normalized_value = {}
+    for arg_key, arg_schema in handler_args_schemas.items():
+
+        if arg_key not in handler_args or handler_args[arg_key] is None:
```

Comment:
Do we need the `handler_args[arg_key] is None` check here?

Reply:
Yes we are expecting some of the values to be None, like in the post request of AdminRoleHandler there are 3 args (username, role, topic_id). If the updated role in frontend is not set to TOPIC_MANAGER then topic_id is automatically set to None. Thus to ignore validation of these cases this extra condition is required.